lib/publiccloud/provider.pm

```diff
@@ -466,6 +466,24 @@ sub terraform_destroy {
     record_info('ERROR', 'Terraform exited with ' . $ret, result => 'fail') if ($ret != 0);
 }
 
+sub terraform_param_tags
+{
+    my ($self) = @_;
+    my $tags = {
+        openqa_ttl        => get_var('MAX_JOB_TIME', 7200) + get_var('PUBLIC_CLOUD_TTL_OFFSET', 300),
+        openqa_var_JOB_ID => get_current_job_id(),
+        map { 'openqa_var_' . $_ => get_var($_) } qw(NAME TEST VERSION FLAVOR DISTRI BUILD),
+    };
+
+    return encode_json($tags);
+}
+
+sub escape_single_quote {
+    my $s = shift;
+    $s =~ s/'/'"'"'/g;
+    return $s;
+}
```

Comment:
This feel too generic to be here. Isn't there a library for this?

Reply:
unfortunately not 
```bash
$ grep -rn  "'"'"'"'"'"'"'"
lib/publiccloud/instance.pm:63:        $cmd =~ s/'/'"'"'/g;
lib/publiccloud/provider.pm:488:    $s =~ s/'/'"'"'/g;
lib/utils.pm:1753:        $value =~ s/'/'"'"'/g;
lib/utils.pm:1755:        $key   =~ s/'/'"'"'/g;
tests/wicked/before_test.pm.orig:33:    my $escaped                = $enable_command_logging =~ s/'/'"'"'/gr;
tests/wicked/before_test.pm:34:    my $escaped                = $enable_command_logging =~ s/'/'"'"'/gr;
```